README.md

```diff
@@ -1,53 +1,294 @@
+
+```yaml
+version: '2'
+
+volumes:
+  apps:
+  config:
+  data:
+  db:
+
+services:
+  db:
+    image: mariadb
+    restart: always
+    volumes:
+      - db:/var/lib/mysql
+    environment:
+      - MYSQL_ROOT_PW=...
+      - MYSQL_USER_PW=...
+      - MYSQL_DATABASE=nextcloud
+      - MYSQL_USER=nextcloud
+
+  app:
+    image: nextcloud
+    links:
+      - db
+    volumes:
+      - data:/var/www/html/data
+      - config:/var/www/html/config
+      - apps:/var/www/html/custom_apps
+    restart: always
+
+  web:
+    image: nginx
+    ports:
+      - "80:80"
+    links:
+      - app
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+    volumes-from:
+      - app
+    restart: always
+```
+
+## SSL encryption
+Until here, we haven't talked about encrypting the connection between your nextcloud host and the clients. Using up-to-date encryption is mandatory if your host is reachable from the internet. There are many different possibilities to introduce encryption. 
+
+An easy and free way to get certificates that are accepted by the browsers is [Let's Encrypt](https://letsencrypt.org/). The whole certificate generation / validation is fully automated and certificate renewals are also very easy. 
+To integrate Let's Encrypt, we recommend using a reverse proxy in front of our nextcloud installation. Your nextcloud will only be reachable through the proxy, which encrypts all traffic to the clients. You can mount your manually generated certificates to the proxy or use a fully automated solution, which generates and renews the certificates for you.
+
+See our [examples](https://github.com/nextcloud/docker/tree/master/.examples) section to get an idea how it works.
+
+# First use
+When you first access your nextcloud, the setup wizard will appear and ask you to choose an administrator account, password and the database connection. For the database use `db` as host and `nextcloud` as table and user name. Also enter the password you chose in your `docker-compose.yml` file.
+
+# Update to a newer version
+Updating the nextcloud container is done by pulling the new image and throwing away the old container. Since all data is stored in volumes nothing gets lost. The startup script will check for the version in your data and the installed docker version. If it finds a mismatch, it automatically starts the upgrade process. Don't forget to add all the volumes to your new container, so it works as expected. 
+
+When the new container starts it detects the mismatch between the installed version of the nextcloud source files and your persistant data. It then unpacks the new sources and starts the upgrade process.
 
-To update your Nextcloud version you simply have to pull and start the new container.
 ```console
 $ docker pull nextcloud
+$ docker stop <your_nextcloud_container>
+$ docker rm <your_nextcloud_container>
 $ docker run -d nextcloud
```

Comment:
@tilosp I'm a bit confused right now. Do you mean the upgrade script needs the old installation to perform a propper upgrade?

What would happen if I just mount the folders listed below and then pull a newer image?
```
$ docker run -d nextcloud \
-v apps:/var/www/html/custom_apps \
-v config:/var/www/html/config \
-v data:/var/www/html/data \
-v theme:/var/www/html/themes/<YOUR_CUSTOM_THEME>
```

I think this:
It would not find any `version.php ` and perform a "new installation" wich copies all the installation files but does not overwrite the existing data files. 

The only problem might be, that occ:upgrade is not triggered. 

We could fix this in the upgrade script or by simply remove suggesting fine grained volume control from the readme.

Have I missed something?